src/main/webapp/app/shared/util/firebase/firebase-review-utils.spec.ts

```diff
@@ -0,0 +1,548 @@
+import { GENE_TYPE, ReviewAction } from 'app/config/constants/firebase';
+import {
+  EditorReviewMap,
+  MetaReviewLevel,
+  ReviewLevel,
+  ReviewLevelType,
+  buildCancerTypeNameReview,
+  buildNameReview,
+  buildRCTReview,
+  buildStringReview,
+  clearAllNestedReviews,
+  findReviewRecursive,
+  findReviews,
+  getRelevantKeysFromUuidKey,
+  getReviewAction,
+  joinPathParts,
+} from './firebase-review-utils';
+import { Gene, GeneType, Implication, Mutation, Review, Treatment, Tumor } from 'app/shared/model/firebase/firebase.model';
+import { DiffMethod } from 'react-diff-viewer-continued';
+import { IDrug } from 'app/shared/model/drug.model';
+
+describe('Firebase Review Utils', () => {
+  describe('getRelevantKeysFromUuidKey', () => {
+    it('should return correct keys', () => {
+      expect(getRelevantKeysFromUuidKey('summary_uuid')).toStrictEqual({
+        fieldKey: 'summary',
+        reviewKey: 'summary_review',
+        uuidKey: 'summary_uuid',
+      });
+    });
+  });
+
+  describe('joinPathParts', () => {
+    it('should join path when parent path is empty', () => {
+      expect(joinPathParts('', 'mutations', '0', 'name')).toEqual('mutations/0/name');
+    });
+
+    it('should join path', () => {
+      expect(joinPathParts('gene', 'mutations', '0', 'name')).toEqual('gene/mutations/0/name');
+    });
+  });
+
+  describe('getReviewAction', () => {
+    test.each([
+      [{ updatedBy: 'User', updateTime: 1 }, 'mutations/0/name_review', ReviewAction.NAME_CHANGE],
+      [{ updatedBy: 'User', updateTime: 1 }, 'mutations/0/tumors/0/cancerTypes_review', ReviewAction.NAME_CHANGE],
+      [{ updatedBy: 'User', updateTime: 1 }, 'mutations/0/tumors/0/TIs/0/treatments/0/name_review', ReviewAction.NAME_CHANGE],
+      [{ updatedBy: 'User', updateTime: 1, promotedToMutation: true }, 'mutations/0/name_review', ReviewAction.PROMOTE_VUS],
+      [{ updatedBy: 'User', updateTime: 1, added: true }, 'mutations/0/name_review', ReviewAction.CREATE],
+      [{ updatedBy: 'User', updateTime: 1, demotedToVus: true }, 'mutations/0/name_review', ReviewAction.DEMOTE_MUTATION],
+      [{ updatedBy: 'User', updateTime: 1, removed: true }, 'mutations/0/name_review', ReviewAction.DELETE],
+      [{ updatedBy: 'User', updateTime: 1, lastReviewed: 'old' }, 'mutations/0/description_review', ReviewAction.UPDATE],
+    ])('should return correct review action', (review, path, expected) => {
```

Comment:
I don't know the answer to this, but if one of these tests breaks will jest tell you which case broke? If it just gives you the test name I think you are able to use test.each to change the test name based on the data passed to it.

Reply:
It gives the test name, but in this case the names are all the same: `should return correct review action`

Reply:
Looks like you can string format a [name](https://jestjs.io/docs/api#1-testeachtablename-fn-timeout) for each test based on the parameters passed to the test case